src/EditorFeatures/CSharpTest/Classification/TotalClassifierTests.cs

```diff
@@ -3022,5 +3028,118 @@ public async Task TestFunctionPointer2(TestHost testHost)
                 Local("fp"),
                 Punctuation.Semicolon);
         }
+
+        [WpfFact]
+        public async Task TestTotalClassifier()
+        {
+            using var workspace = TestWorkspace.CreateCSharp(""""
+                using System.Text.RegularExpressions;
+
+                class C
+                {
+                    // class D { }
+                    void M()
+                    {
+                        new Regex("(a)");
+                        var s1 = "s1";
+                        var s2 = $"s2";
+                        var s3 = @"s3";
+                        var s4 = """
+                        s4
+                        """;
+                    }
+                }
+                """");
+            var document = workspace.Documents.First();
+
+            var listenerProvider = workspace.ExportProvider.GetExportedValue<IAsynchronousOperationListenerProvider>();
+            var globalOptions = workspace.ExportProvider.GetExportedValue<IGlobalOptionService>();
+
+            var provider = new TotalClassificationTaggerProvider(
+                workspace.GetService<IThreadingContext>(),
+                workspace.GetService<ClassificationTypeMap>(),
+                globalOptions,
+                visibilityTracker: null,
+                listenerProvider);
+
+            var buffer = document.GetTextBuffer();
+            using var tagger = provider.CreateTagger(document.GetTextView(), buffer);
+
+            var waiter = listenerProvider.GetWaiter(FeatureAttribute.Classification);
+            await waiter.ExpeditedWaitAsync();
+
+            var allCode = buffer.CurrentSnapshot.GetText();
+            var tags = tagger!.GetTags(new NormalizedSnapshotSpanCollection(buffer.CurrentSnapshot.GetFullSpan()));
+
+            var actualOrdered = tags.OrderBy((t1, t2) => t1.Span.Span.Start - t2.Span.Span.Start);
```

Comment:
>tags.OrderBy((t1, t2) => t1.Span.Span.Start - t2.Span.Span.Start); [](http://example.com/codeflow?start=32&length=66)

any concern about items with the same start being ordered inconsistently? (ie, should it also compare end, classificationtype to ensure consistent ordering?)